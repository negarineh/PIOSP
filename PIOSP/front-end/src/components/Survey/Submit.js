import React from 'react';
import PropTypes from 'prop-types';

import '../stylesheets/Survey.css';
import '../stylesheets/MainPage.css';
import '../stylesheets/Submit.css';
import MuiThemeProvider from 'material-ui/styles/MuiThemeProvider';
import { Segment, Step} from 'semantic-ui-react';

import cookie from 'react-cookies';
import {Link} from 'react-router-dom';

const Star = ({ selected=false, onClick=f=>f }) =>
        <div className={(selected) ? "star selected" : "star"}
             onClick={onClick}>
        </div>


// after participant answers questions, survey will redirect to this page for last part
// in this page participants rate their experience about survey and give information about their pollination research
class Submit extends React.Component {
  constructor(props) {
    super(props);   
    this.state={
      experience: '',
      description: '',
      participation:'',
      photos: [],
      items: [],
      postItem: [],
      finished: false,
      stepIndex: 0,
      starsSelected: props.starsSelected || 0
    };
    this.addItem = this.addItem.bind(this);
    this.change = this.change.bind(this)
  }

  componentDidMount() {

    window.setTimeout(function(){

    // Move to a new location or you can do something else
    window.location.href = "http://localhost:3001/finish";
    cookie.remove('jwt', { path: '/' });
    }, 5 * 60 * 1000);
  }

  change(starsSelected) {
      this.setState({starsSelected})
  }

  addItem(e) {

    e.preventDefault();
    const data = new FormData(e.target);
    
    var curDescription = data.get('description'),
        curparticipation = data.get('participation'),
        curUserId =  cookie.load('jwt');

    var itemArray = this.state.items;
    var alwaysTrue = true;
    
    if (alwaysTrue) {
    
            // On submit of the form, send a POST request with the data to the server.
          fetch(`saveExperience`,{
            method: 'POST',
            body:JSON.stringify({
              id: curUserId,
              experience: this.state.starsSelected,
              description: curDescription,
              participation: curparticipation
            }),
            headers: {"Content-Type": "application/json"}
          }).then(function(response){
            if (response.status >= 200 && response.status < 300) {
              return response;
            }
            const error = new Error(`HTTP Error ${response.statusText}`);
            error.status = response.statusText;
            error.response = response;
            console.log(error); // eslint-disable-line no-console
            throw error;
          }).catch(err => console.log(err));

      this.setState({
        items: itemArray,
        sendItems: itemArray
      });
    }
    
    this.description.value = "";

    this.setState({
      number: this.state.number+1
    })

    cookie.remove('jwt', { path: '/' });

    console.log(itemArray); // eslint-disable-line no-console
    console.log(this.state.number); // eslint-disable-line no-console
   
  }
  
  render() {

    let experience = this.state.experience;

    const { photos } = this.state;
    const {totalStars} = this.props
    const {starsSelected} = this.state

    return (
      <div className="mainpage">
        <MuiThemeProvider>
        {/* <Header/> */}
        {!experience &&
          <div className="wrappermainpage">      
             <div style={{width: 'auto', margin: 'auto', color:"dimgray", paddingLeft:"10px", overflow:'auto'}}>
             <Step.Group ordered size="mini" >
              <Step completed >
                <Step.Content>
                  <Step.Title>Enrolling</Step.Title>
                  <Step.Description>Insert your email </Step.Description>
                </Step.Content>
              </Step>

              <Step completed >
                <Step.Content>
                  <Step.Title>Confirmed Email </Step.Title>
                  <Step.Description>Welcome to our Survey</Step.Description>
                </Step.Content>
              </Step>

              <Step completed >
                <Step.Content>
                  <Step.Title>Survey Questions</Step.Title>
                  <Step.Description>Please answer questions</Step.Description>
                </Step.Content>
              </Step>

              <Step active >
                <Step.Content>
                  <Step.Title>Rating and Further Experiences</Step.Title>
                  <Step.Description>Please rate our survey</Step.Description>
                </Step.Content>
              </Step>

              <Step disabled >
                <Step.Content>
                  <Step.Title>Survey Submitted</Step.Title>
                  <Step.Description>Thanks for your participation</Step.Description>
                </Step.Content>
              </Step>
            </Step.Group>
            </div>

        <Segment style={{margin:"10px"}}>
          <form  onSubmit={this.addItem}> 
            <p key="a" className="psubmit" > 
              Please rate your insect identification experience, 
              where 1 is have no experience and 5 is highly experienced :
            </p>
              <div key="b" className="questionsubmit">
                <div className="react-container">
                  <div className="star-rating">
                      {[...Array(totalStars)].map((n, i) =>
                          <Star key={i}
                                selected={i<starsSelected}
                                onClick={() => this.change(i+1)}
                                style={{display:"inline", margin:"20px"}}
                                ref={(a) => this.experience = a}
                                name="experience" 
                                value = {starsSelected}
                          />
                      )}
                  </div>
                  </div>
              </div>
            <p className="psubmit" > Have you ever participated in a citizen science project?
            </p>
              <div className="form-group-sub">
                <div className="form-group-submit">
                <label className="labelpage1">
                  <input 
                    //  placeholder= "Email"
                    type="radio" 
                    name="participation" 
                    className="form-control-submit"
                    value="Yes"
                    ref={(a) => this.participation = a}
                    required />
                    Yes
                      </label>
                </div>
                <div className="form-group-submit">
                <label className="labelpage1">
                  <input 
                    //  placeholder= "Email"
                    type="radio" 
                    name="participation" 
                    className="form-control-submit"
                    value="No"
                    ref={(a) => this.participation = a}
                    required />
                    No
                  </label>
                </div>
              </div>
              
              <p key="d" className="psubmit"  > Please give us brief description of type of project:
              </p>
                <textarea 
                  //  placeholder= "Email"
                   type="input" 
                   name="description" 
                   className="form-control-description"
                   ref={(a) => this.description = a}
                  //  maxLength = {40}
                   />
              <button 
                type="submit" 
                className="btn btn-primary2"
                >
                <Link to={'/finish'}>Submit</Link>
              </button>
            </form>
        </Segment>
            {photos} 
          </div>
  }
        </MuiThemeProvider>
      </div>
    );
  }
  }

  Submit.defaultProps = {
    starsSelected: 0,
    totalStars: 5,
    history: {},
    addItem: () => {},
    change: () => {}
  }

  Submit.propTypes = {
    starsSelected: PropTypes.number.isRequired,
    totalStars: PropTypes.number.isRequired,
    history: PropTypes.object.isRequired,
    addItem: PropTypes.func.isRequired,
    change: PropTypes.func.isRequired,
};

export default Submit;