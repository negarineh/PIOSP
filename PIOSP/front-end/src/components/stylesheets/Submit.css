.form-control-submit {
  margin-bottom: 10px; }

.btn-primary2 {
  position: relative;
  display: block;
  margin-left: 10px;
  padding: 10px 30px;
  font-size: 20px;
  font-family: "fantasy";
  font-weight: 400;
  color: white;
  background-image: linear-gradient(#20d31a, #8ac208);
  border: none;
  z-index: 1;
  border-radius: 5px; }
  .btn-primary2:before {
    position: absolute;
    content: '';
    top: 0;
    left: 0;
    width: 0;
    height: 100%;
    background-image: linear-gradient(#0a830a, #21ca0a);
    transition: width 250ms;
    z-index: -1;
    border-radius: 5px; }
  .btn-primary2:hover {
    color: white;
    border-radius: 5px; }
    .btn-primary2:hover:before {
      width: 100%; }

.labelpage1 {
  color: #463831;
  font-family: sans-serif;
  font-size: 15px; }

.psubmit {
  float: left;
  align-items: left;
  height: 50%;
  width: 50%;
  color: #463831;
  line-height: 1.5;
  font-family: sans-serif;
  font-size: 15px;
  text-justify: auto;
  padding-left: 10px;
  clear: none; }

.form-control-submit {
  display: inline-block; }

.form-group-submit {
  display: inline-block; }

.form-group-sub {
  display: inline-block;
  padding-left: 100px;
  padding-top: 20px;
  padding-bottom: 25px; }

.fieldsetsubmit {
  text-align: left;
  padding: 20px;
  padding-bottom: 20px;
  margin-top: 20px;
  margin-left: 20px;
  margin-right: 20px;
  margin-bottom: 20px;
  border-radius: 10px; }

.legendsubmit {
  font-family: "fantasy";
  font-size: 20px; }

.imgsubmit {
  width: 500px;
  height: 400px;
  float: left;
  clear: none;
  border-style: double;
  border-radius: 10px; }

.questionsubmit {
  display: inline;
  font-family: sans-serif;
  font-size: 15px; }

.form-control-description {
  width: 270px;
  height: 100px;
  font-family: "fantasy";
  font-size: 20px;
  border-radius: 5px;
  display: inline-block;
  margin-top: 30px;
  margin-left: 92px;
  resize: none; }

.h2submit {
  text-align: center;
  font-family: sans-serif; }

.react-container {
  display: inline-block;
  align-items: center;
  justify-content: center;
  padding-left: 80px;
  padding-bottom: 25px; }

.star-rating {
  display: inline-block;
  flex-wrap: wrap;
  justify-content: space-around; }

.star {
  cursor: pointer;
  width: 2em;
  height: 2em;
  background-color: grey;
  -webkit-clip-path: polygon(50% 0%, 63% 38%, 100% 38%, 69% 59%, 82% 100%, 50% 75%, 18% 100%, 31% 59%, 0% 38%, 37% 38%);
  clip-path: polygon(50% 0%, 63% 38%, 100% 38%, 69% 59%, 82% 100%, 50% 75%, 18% 100%, 31% 59%, 0% 38%, 37% 38%);
  display: inline-block;
  margin: 15px; }

.star.selected {
  background-image: linear-gradient(#fbff0a, #fadc30f3); }

.submitsurvey {
  display: inline-block;
  color: dimgray; }

.divp {
  display: inline-block; }
