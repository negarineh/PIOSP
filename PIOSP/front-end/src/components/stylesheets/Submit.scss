 // Variables
 $color-bg: white;
 $color-main: #333;
 $color-hover: lighten(#0275d8, 1%);
 $padding: 20px;
 $radius: 10px;
 $accent-color: #dfcfad; 
 $font-size:20px;
 $font-family: 'fantasy';
 $background-color: rgb(78, 82, 76);
 $background-color-after: #e74c3c;
 
 // Mixins
 @mixin transition($prop: all, $duration: 0.3s, $timing: ease-in-out) {
   transition: $prop $duration $timing;
 }
  
 
 // h1 {
 //   color: $color-main;
 //   font-size: 30px;
 //   margin-top: 0;
 //   text-align: center; 
 //   font-family: 'fantasy';
 // }
  
 .form-control-submit { 
 //   border: none;
 //   border-bottom: 1px solid $color-main;
 //   background: none;
 //   width: 100%;
 //   font-size: 18px;
 //   font-weight: bold;
 //   line-height: 1.2;
 //   outline: none;
 //   text-align: right;
 //   border-radius: $radius;
 //   vertical-align: bottom;
 //   position: relative;
 //   // float: left;
 //   // *overflow: hidden;
     margin-bottom: 10px;
 }
 // .btn:hover {
 //   background: $color-hover;
 //   border-color: $color-hover;
 // }
    
 // .wrapperpage1 {      
 //   margin: 50px;
 //   font-size: 20px;
 //   // text-align: center;
 //   width: 1200px; 
 //   height: 550px;
 //   border-radius: $radius;
 //   background: transparent;
 //   box-shadow: 0 15px 35px rgba(28, 28, 204, 0.1), 0 5px 15px rgba(0, 0, 0, 0.514);
 //   // padding: $padding;
 //   background-color: white;
 //   }  
 
 .btn-primary2{
             position: relative;
             display: block;
            //  margin: 30px auto 0;
            margin-left: 10px;
             padding: 10px 30px;
             font-size: $font-size;
             font-family: $font-family;
             font-weight: 400;
             color: white;
             background-image: linear-gradient(#20d31a, #8ac208);
             border: none;
             z-index: 1;
             border-radius: 5px;
             &:before {
                 position: absolute;
                 content: '';
                 top: 0;
                 left: 0;
                 width: 0;
                 height: 100%;
                 background-image: linear-gradient(#0a830a,#21ca0a);
                 transition: width 250ms;
                 z-index: -1;
                 border-radius: 5px;
             }
             
             &:hover {
                 color: white;
                 border-radius: 5px;
                 
                 &:before {
                     width: 100%;
                 }
             }
   }
 
 .labelpage1{
   color: rgb(70, 56, 49);
   // background: $accent-color;
   font-family: sans-serif;
   font-size: 15px;
 //   display: block; 
//  float: left;
//   clear: none;
 } 
  
 .psubmit{
  //  display: inline-block;
  float: left;
   align-items: left ;
 //   justify-content: center;
   // position: absolute;
   height: 50%; 
   width: 50%;
   color: rgb(70, 56, 49);
  //  padding-bottom: $padding;
   line-height: 1.5;
   font-family: sans-serif;
   font-size: 15px;
   text-justify: auto;
   padding-left: 10px;
   clear: none;
 }
 
 .form-control-submit
 {
   display: inline-block;
 }

 .form-group-submit{
  display: inline-block;
 }

 .form-group-sub{
   display: inline-block;
  //  float: left;
  //  clear: none;
  padding-left: 100px;
  padding-top: 20px;
  padding-bottom: 25px;
 }

 .fieldsetsubmit{
   text-align: left;
   padding: $padding; 
   padding-bottom: 20px; 
   margin-top: $padding;  
   margin-left: $padding;
   margin-right: $padding;
   margin-bottom: $padding;
   border-radius: $radius; 
 }
 
 .legendsubmit{
     // padding-top: 10px;
     font-family: $font-family;
     font-size: $font-size;
     
 }
 
 // .wrapper1.input{
 //   align-self: auto;
 //   float: left;
 // }
 // .wrapper1.label{
 //   text-align: right;
 //   float: left;
 // }
 
 // .form-group{
 //   padding: 5px 5px 5px 5px;
 //   left: 60px;
 //   position: absolute;
 // }
 
 .imgsubmit{
    width:500px;
    height:400px;
    float: left;
    clear: none;
    border-style: double;
    border-radius: $radius;
  }

 .questionsubmit{
   display: inline;
   font-family: sans-serif;
   font-size: 15px;
 }

 .form-control-description{
  width:270px;
  height:100px;
  font-family: "fantasy";
  font-size: 20px;
  border-radius: 5px;
  display: inline-block;
  margin-top: 30px;
  margin-left: 92px;
  resize: none;
} 

.h2submit{
    text-align: center;
    font-family: sans-serif;
    // font-size: 15px;
}

.react-container {
  display: inline-block;
  align-items: center;
  justify-content: center;
  padding-left: 80px;
  padding-bottom: 25px;
}

.star-rating {
  display: inline-block;
  flex-wrap: wrap;
  justify-content: space-around;
}

.star {
  cursor: pointer;
  width: 2em;
  height: 2em;
  background-color: grey;
  -webkit-clip-path:
    polygon(50% 0%, 63% 38%, 100% 38%,
      69% 59%, 82% 100%, 50% 75%,
      18% 100%, 31% 59%, 0% 38%, 37% 38%);
  clip-path: polygon(50% 0%, 63% 38%,
    100% 38%, 69% 59%, 82% 100%,
    50% 75%, 18% 100%, 31% 59%,
    0% 38%, 37% 38%);
    display: inline-block;
    margin: 15px;
}

.star.selected {
  background-image: linear-gradient(#fbff0a,#fadc30f3);
}


.submitsurvey{
  display: inline-block;
  color: dimgray
}

.divp{
  // float: left;
  // clear: none;
  display: inline-block;
}