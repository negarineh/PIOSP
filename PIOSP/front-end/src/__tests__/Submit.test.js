import React from "react";
import Enzyme, { shallow } from "enzyme";
import assert from "assert";
import Submit from "../components/Survey/Submit";
import Adapter from 'enzyme-adapter-react-16';
import toJson from 'enzyme-to-json';
 
Enzyme.configure({ adapter: new Adapter() });

// unit tests for the UserDeletePrompt component
describe('Submit component', () => {
  describe('render()', () => {
    it('should render the component', () => {
      const props = {starSelected: 0, totalStars: 5, history: {}, addItem: ()=>{}, change: ()=>{}};
      const wrapper = shallow(<Submit {...props}/>);
      const tree = toJson(wrapper);
      expect(tree).toMatchSnapshot();
      assert.equal(wrapper.length, 1);
    });
  });
});
