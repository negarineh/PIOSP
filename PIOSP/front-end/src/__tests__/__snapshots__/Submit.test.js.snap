// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Submit component render() should render the component 1`] = `
<div
  className="mainpage"
>
  <MuiThemeProvider>
    <div
      className="wrappermainpage"
    >
      <div
        style={
          Object {
            "color": "dimgray",
            "margin": "auto",
            "overflow": "auto",
            "paddingLeft": "10px",
            "width": "auto",
          }
        }
      >
        <StepGroup
          ordered={true}
          size="mini"
        >
          <Step
            completed={true}
          >
            <StepContent>
              <StepTitle>
                Enrolling
              </StepTitle>
              <StepDescription>
                Insert your email 
              </StepDescription>
            </StepContent>
          </Step>
          <Step
            completed={true}
          >
            <StepContent>
              <StepTitle>
                Confirmed Email 
              </StepTitle>
              <StepDescription>
                Welcome to our Survey
              </StepDescription>
            </StepContent>
          </Step>
          <Step
            completed={true}
          >
            <StepContent>
              <StepTitle>
                Survey Questions
              </StepTitle>
              <StepDescription>
                Please answer questions
              </StepDescription>
            </StepContent>
          </Step>
          <Step
            active={true}
          >
            <StepContent>
              <StepTitle>
                Rating and Further Experiences
              </StepTitle>
              <StepDescription>
                Please rate our survey
              </StepDescription>
            </StepContent>
          </Step>
          <Step
            disabled={true}
          >
            <StepContent>
              <StepTitle>
                Survey Submitted
              </StepTitle>
              <StepDescription>
                Thanks for your participation
              </StepDescription>
            </StepContent>
          </Step>
        </StepGroup>
      </div>
      <Segment
        style={
          Object {
            "margin": "10px",
          }
        }
      >
        <form
          onSubmit={[Function]}
        >
          <p
            className="psubmit"
            key="a"
          >
            Please rate your insect identification experience, where 1 is have no experience and 5 is highly experienced :
          </p>
          <div
            className="questionsubmit"
            key="b"
          >
            <div
              className="react-container"
            >
              <div
                className="star-rating"
              >
                <Star
                  key="0"
                  name="experience"
                  onClick={[Function]}
                  selected={false}
                  style={
                    Object {
                      "display": "inline",
                      "margin": "20px",
                    }
                  }
                  value={0}
                />
                <Star
                  key="1"
                  name="experience"
                  onClick={[Function]}
                  selected={false}
                  style={
                    Object {
                      "display": "inline",
                      "margin": "20px",
                    }
                  }
                  value={0}
                />
                <Star
                  key="2"
                  name="experience"
                  onClick={[Function]}
                  selected={false}
                  style={
                    Object {
                      "display": "inline",
                      "margin": "20px",
                    }
                  }
                  value={0}
                />
                <Star
                  key="3"
                  name="experience"
                  onClick={[Function]}
                  selected={false}
                  style={
                    Object {
                      "display": "inline",
                      "margin": "20px",
                    }
                  }
                  value={0}
                />
                <Star
                  key="4"
                  name="experience"
                  onClick={[Function]}
                  selected={false}
                  style={
                    Object {
                      "display": "inline",
                      "margin": "20px",
                    }
                  }
                  value={0}
                />
              </div>
            </div>
          </div>
          <p
            className="psubmit"
          >
             Have you ever participated in a citizen science project?
          </p>
          <div
            className="form-group-sub"
          >
            <div
              className="form-group-submit"
            >
              <label
                className="labelpage1"
              >
                <input
                  className="form-control-submit"
                  name="participation"
                  required={true}
                  type="radio"
                  value="Yes"
                />
                Yes
              </label>
            </div>
            <div
              className="form-group-submit"
            >
              <label
                className="labelpage1"
              >
                <input
                  className="form-control-submit"
                  name="participation"
                  required={true}
                  type="radio"
                  value="No"
                />
                No
              </label>
            </div>
          </div>
          <p
            className="psubmit"
            key="d"
          >
             Please give us brief description of type of project:
          </p>
          <textarea
            className="form-control-description"
            name="description"
            type="input"
          />
          <button
            className="btn btn-primary2"
            type="submit"
          >
            <Link
              replace={false}
              to="/finish"
            >
              Submit
            </Link>
          </button>
        </form>
      </Segment>
    </div>
  </MuiThemeProvider>
</div>
`;
